lib/resourcemerge/machineconfig.go

```diff
@@ -41,6 +41,20 @@ func EnsureMachineConfigPool(modified *bool, existing *mcfgv1.MachineConfigPool,
 		*modified = true
 		existing.Spec.MachineSelector = required.Spec.MachineSelector
 	}
+
+	if existing.Spec.MachineSelector != nil {
+		existing.Spec.NodeSelector = existing.Spec.MachineSelector
+		required.Spec.NodeSelector = required.Spec.MachineSelector
```

Comment:
this line isn't required, as that's what's coming from `./manifests/...` and we drive that (to the new one already)

Reply:
actiually, this whole if statement is not needed as it's only a new payload with nodeSelector when we merge this PR. We need to change the `machineSelector`in stuff like this indeed https://github.com/openshift/machine-config-operator/blob/76232a00900850733d05b049d82683cbdb1401e0/manifests/master.machineconfigpool.yaml#L11